### ISSUE TITLE
Changes in html mode are not saved

### ISSUE THREAD

**Issue by hypest** [wordpress-mobile/gutenberg-mobile#1572]
Changing the post content via the html mode are not saved when exiting the post.

**To Reproduce**
1. Create or open a post in WPAndroid or WPiOS
2. Switch to html mode via the 3-dot menu
3. Make a trivial change to content without malforming. For example, add more text to a paragraph block.
4. Tap back (or "X" on iOS) to exit the editor
5. Nothing seems to happen (say, uploading changes to the post)
6. Open the post again and notice that the changes are not there 💥 

**Expected behavior**
The post should have been marked as changed and saved locally and/or remotely. Opening the post again should show the edits.

**Smartphone:**
 - Device: Pixel 2 XL
 - OS: Android 9
 - Version WPAndroid 13.6-rc-1

and

 - Device: iPhone X Simulator
 - OS: iOS 13.1
 - Version WPiOS 13.6 (DEV)

**Comment by rezzap** [comment]
This was reported again in 2533079-zen

I replicated this by following these steps:

1. Made a post through the browser using block editor with a table block.
2. Published the post.
3. Opened the post in the app to edit and went to the HTML mode.
4. Made changes in the table without changing any of the HTML formattings. 
5. Updated post (advised changes were published)
6. View post and changes are not showing.
7. Go back to post editor, changes did not save.

When I tried from the browser, this updated accordingly. 

Tried in both Android (Samsung TabA, OS: Android 9, WPAndroid 13.6) and iOS (iPhone X, iOS 13.2.3, WPiOS 13.7)
